api/server/server.go

```diff
@@ -899,14 +899,27 @@ func (s *Server) getImagesGet(eng *engine.Engine, version version.Version, w htt
 	if version.GreaterThan("1.0") {
 		w.Header().Set("Content-Type", "application/x-tar")
 	}
-	var job *engine.Job
+	output := utils.NewWriteFlusher(w)
+
+	imageExportConfig := &graph.ImageExportConfig{
+		Engine:    eng,
+		Outstream: output,
+	}
 	if name, ok := vars["name"]; ok {
-		job = eng.Job("image_export", name)
+		imageExportConfig.Name = append(imageExportConfig.Name, name)
 	} else {
-		job = eng.Job("image_export", r.Form["names"]...)
+		imageExportConfig.Name = r.Form["names"]
 	}
-	job.Stdout.Add(w)
-	return job.Run()
+
+	if err := s.daemon.Repositories().ImageExport(imageExportConfig); err != nil {
+		if !output.Flushed() {
```

Comment:
Here is an attempt to write to the `OutStream`, simply return error would cause an error somehow like `http: multiple response.WriteHeader calls` to my best knowledge. Exactly the same case as `pull`, `import` and `push`. 
https://github.com/docker/docker/blob/master/graph/export.go#L99
The old one with job is ok without `Flushed` or sth.